llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -17660,6 +17658,14 @@ static SDValue tryCombineToBSL(SDNode *N, TargetLowering::DAGCombinerInfo &DCI,
     for (int j = 1; j >= 0; --j) {
       BuildVectorSDNode *BVN0 = dyn_cast<BuildVectorSDNode>(N0->getOperand(i));
       BuildVectorSDNode *BVN1 = dyn_cast<BuildVectorSDNode>(N1->getOperand(j));
```

Comment:
These casts are not used by the new code so really belong just before their null checks.

Reply:
Done.